media-libs/libsdl2/patches/libsdl2-2.0.20.patchset

```diff
@@ -1265,3 +1265,145 @@ index c5026de..868512e 100644
+         if (SDL_AppThread == NULL) {
+             return SDL_SetError("Couldn't create BApplication thread");
+         }
++    	}
+ 
+         /* Change working directory to that of executable */
+         app_info info;
+-- 
+2.30.2
+
+
+From 31f33ea16f1d547150cf06f533f6c0ffb2d3b058 Mon Sep 17 00:00:00 2001
+From: Ivan Holmes <ivan@ivanholmes.co.uk>
+Date: Sat, 7 May 2022 22:10:49 +0100
+Subject: Convert SDL_BApp members to static
+
+
+diff --git a/src/main/haiku/SDL_BApp.h b/src/main/haiku/SDL_BApp.h
+index 215f836..6c59ad0 100644
+--- a/src/main/haiku/SDL_BApp.h
++++ b/src/main/haiku/SDL_BApp.h
+@@ -83,12 +83,7 @@ enum ToSDL {
+ class SDL_BApp : public BApplication {
+ public:
+     SDL_BApp(const char* signature) :
+-        BApplication(signature) {
+-#if SDL_VIDEO_OPENGL
+-        _current_context = NULL;
+-#endif
+-    }
+-
++        BApplication(signature) {}
+ 
+     virtual ~SDL_BApp() {
+     }
+@@ -179,7 +174,7 @@ public:
+     }
+ 
+     /* Window creation/destruction methods */
+-    int32 GetID(SDL_Window *win) {
++    static int32 GetID(SDL_Window *win) {
+         int32 i;
+         for(i = 0; i < _GetNumWindowSlots(); ++i) {
+             if( GetSDLWindow(i) == NULL ) {
+@@ -200,19 +195,19 @@ public:
+ 
+     /* FIXME: Bad coding practice, but I can't include SDL_BWin.h here.  Is
+        there another way to do this? */
+-    void ClearID(SDL_BWin *bwin); /* Defined in SDL_BeApp.cc */
++    static void ClearID(SDL_BWin *bwin); /* Defined in SDL_BeApp.cc */
+ 
+ 
+-    SDL_Window *GetSDLWindow(int32 winID) {
++    static SDL_Window *GetSDLWindow(int32 winID) {
+         return _window_map[winID];
+     }
+ 
+ #if SDL_VIDEO_OPENGL
+-    BGLView *GetCurrentContext() {
++    static BGLView *GetCurrentContext() {
+         return _current_context;
+     }
+     
+-    void SetCurrentContext(BGLView *newContext) {
++    static void SetCurrentContext(BGLView *newContext) {
+         if(_current_context)
+             _current_context->UnlockGL();
+         _current_context = newContext;
+@@ -402,29 +397,29 @@ private:
+ 
+     /* Vector functions: Wraps vector stuff in case we need to change
+        implementation */
+-    void _SetSDLWindow(SDL_Window *win, int32 winID) {
++    static void _SetSDLWindow(SDL_Window *win, int32 winID) {
+         _window_map[winID] = win;
+     }
+ 
+-    int32 _GetNumWindowSlots() {
++    static int32 _GetNumWindowSlots() {
+         return _window_map.size();
+     }
+ 
+ 
+-    void _PopBackWindow() {
++    static void _PopBackWindow() {
+         _window_map.pop_back();
+     }
+ 
+-    void _PushBackWindow(SDL_Window *win) {
++    static void _PushBackWindow(SDL_Window *win) {
+         _window_map.push_back(win);
+     }
+ 
+ 
+     /* Members */
+-    std::vector<SDL_Window*> _window_map; /* Keeps track of SDL_Windows by index-id */
++    static std::vector<SDL_Window*> _window_map; /* Keeps track of SDL_Windows by index-id */
```

Comment:
Looking into things more I actually think we shouldn't be using statics here. Instead we should put these in the private `driverdata` field that is present in `SDL_VideoDevice`, which is passed in as _THIS to all the implementation functions. That may be more work, though, and might require some more memory management. I can work on that if you are unsure how to do it.

Reply:
I am definitely not sure how to do it, but I'll have a look - I won't learn anything otherwise!
Will let you know if/when I decide it is beyond my ability.

Reply:
Okay, I am really not sure how to do this. Apologies for creating work for you...

Reply:
Well, if you want to learn, I can probably help you through it via IRC/Matrix if you want. Otherwise, it's fine, as this was on my TODO list anyway :)

Reply:
Ok, I definitely do want to learn, I'll log on to IRC shortly :)